File: CRS-Group-A-Java-POS/src/com/lti/bean/CourseCatalog.java
package com.lti.bean;

import java.util.ArrayList;

/**
 * @author Sebastian
 *
 */

public class CourseCatalog {
	
	private ArrayList<Course> courses;
	
	private int id;
	private int professorId;
	private int departmentId;
	private String prerequisites;
	private int credits;
	private int capacity;
	private int enrolled;
	private String semester;
		
	public CourseCatalog(int id, int professorId, int departmentId, 
			String prerequisites, int credits, int capacity, int enrolled, String semester) {
		
		this.id = id;
		this.professorId = professorId;
		this.departmentId = departmentId;
		this.prerequisites = prerequisites;
		this.credits = credits;
		this.capacity = capacity;
		this.enrolled = enrolled;
		this.semester = semester;
	}
	
	public CourseCatalog() {
		
		courses = new ArrayList<Course>();
	}
	

	/**
	 * @return the id
	 */
	public int getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(int id) {
		this.id = id;
	}

	/**
	 * @return the professorId
	 */
	public int getProfessorId() {
		return professorId;
	}

	/**
	 * @param professorId the professorId to set
	 */
	public void setProfessorId(int professorId) {
		this.professorId = professorId;
	}

	/**
	 * @return the departmentId
	 */
	public int getDepartmentId() {
		return departmentId;
	}

	/**
	 * @param departmentId the departmentId to set
	 */
	public void setDepartmentId(int departmentId) {
		this.departmentId = departmentId;
	}

	/**
	 * @return the prerequisites
	 */
	public String getPrerequisites() {
		return prerequisites;
	}

	/**
	 * @param prerequisites the prerequisites to set
	 */
	public void setPrerequisites(String prerequisites) {
		this.prerequisites = prerequisites;
	}

	/**
	 * @return the credits
	 */
	public int getCredits() {
		return credits;
	}

	/**
	 * @param credits the credits to set
	 */
	public void setCredits(int credits) {
		this.credits = credits;
	}

	/**
	 * @return the capacity
	 */
	public int getCapacity() {
		return capacity;
	}

	/**
	 * @param capacity the capacity to set
	 */
	public void setCapacity(int capacity) {
		this.capacity = capacity;
	}

	/**
	 * @return the enrolled
	 */
	public int getEnrolled() {
		return enrolled;
	}

	/**
	 * @param enrolled the enrolled to set
	 */
	public void setEnrolled(int enrolled) {
		this.enrolled = enrolled;
	}

	/**
	 * @return the semester
	 */
	public String getSemester() {
		return semester;
	}

	/**
	 * @param semester the semester to set
	 */
	public void setSemester(String semester) {
		this.semester = semester;
	}

	public ArrayList<Course> getCourses() {
		return courses;
	}

	public void setCourses(ArrayList<Course> courses) {
		this.courses = courses;
	}
	
}

File: CRS-GROUP-A-Angular-UI-UX-App/src/app/models/course-catalog.ts
export class CourseCatalog {
    private id: number;
    private professorID: number;
    private departmentID: number;
    private prerequisites: string;
    private credits: number;
    private capacity: number;
    private enrolled: number;
    private semester: string;

    constructor(id: number, professorID: number, departmentID: number, prerequisites: string, credits: number, capacity: number, enrolled: number, semester: string){
        this.id = id;
        this.professorID = professorID;
        this.departmentID = departmentID;
        this.prerequisites = prerequisites;
        this.credits = credits;
        this.capacity = capacity;
        this.enrolled = enrolled;
        this.semester = semester;
    }

    getID(): number{
        return this.id;
    }

    setID(id: number){
        this.id = id;
    }

    getProfessorID(): number{
        return this.professorID;
    }

    setProfessorID(professorID: number){
        this.professorID = professorID;
    }

    getDepartmentID(): number{
        return this.departmentID;
    }

    setDepartmentID(departmentID: number){
        this.departmentID = departmentID;
    }

    getPrerequisites(): string{
        return this.prerequisites;
    }

    setPrerequisites(prerequisites: string){
        this.prerequisites = prerequisites;
    }

    getCredits(): number{
        return this.credits;
    }

    setCredits(credits: number){
        this.credits = credits;
    }

    getCapacity(): number {
		return this.capacity;
	}

	setCapacity(capacity: number) {
		this.capacity = capacity;
	}

	getEnrolled() {
		return this.enrolled;
	}

	setEnrolled(enrolled: number) {
		this.enrolled = enrolled;
	}

	getSemester(): string {
		return this.semester;
	}

	setSemester(semester: string) {
		this.semester = semester;
	}
}